src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

```diff
@@ -57,28 +58,37 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 		return;
 	}
 
+
+	@memoize // Without memoizing this, we'd create thousands of ConnectionProfile objects
+	public canDragToConnectionProfileGroup(source: any, targetConnectionProfileGroup: ConnectionProfileGroup) {
```

Comment:
Hmm, might have an issue if :

1. You start off with two folders at the same level (A and B)
2. Drag B onto A so it gets placed under A (that operation is now memoized as being valid)
3. Drag B out to the root level so it's back on the same level as A
4. Drag A onto B so it gets placed under B
5. Now try to drag B onto A. It should fail because it's an ancestor of A but because of the memoization earlier I believe this will be cached as being a valid move